rero_ils/modules/items/tasks.py

```diff
@@ -23,11 +23,46 @@
 from flask import current_app
 
 from .api import Item
+from .utils import get_provisional_items_pids_candidate_to_delete
 from ..api import IlsRecordError
 from ..holdings.api import Holding
 from ..utils import extracted_data_from_ref, set_timestamp
 
 
+@shared_task()
+def delete_provisional_items(verbose=True):
+    """Delete checked-in provisional items.
+
+    For the list of candidates item pids to delete, this method tries to delete
+    each item. The item will not be deleted if the method `can_delete` is True.
+    The reasons not to delete a provisional item are the same as other items:
+    1. no active loans
+    2. no fees
+
+    :param verbose: is the task should be verbose.
+    """
+    count_items = 0
+    deleted_items = 0
+    for count_items, item_pid in enumerate(
+            get_provisional_items_pids_candidate_to_delete()):
+        try:
```

Comment:
`count_items` computation seems complicated in this way.
```python
count_items, deleted_items = 0, 0
for pid in get_provisional_items_pids_candidate_to_delete(): 
    count_items += 1
    ...
```

Reply:
https://realpython.com/python-enumerate/
Python’s enumerate() lets you write Pythonic for loops when you need a count and the value from an iterable. The big advantage of enumerate() is that it returns a tuple with the counter and value, so you don’t have to increment the counter yourself. It also gives you the option to change the starting value for the counter.